evm_loader/program/src/transaction.rs

```diff
@@ -120,24 +120,16 @@ impl rlp::Decodable for UnsignedTransaction {
     }
 }
 
-pub fn get_data(raw_tx: &[u8]) -> (u64, Option<H160>, Vec<u8>) {
-    let tx: Result<UnsignedTransaction, _> = rlp::decode(&raw_tx);
-    let tx = tx.unwrap();
-
-    (tx.nonce, tx.to, tx.call_data)
-}
-
-pub fn verify_tx_signature(signature: &[u8], unsigned_trx: &[u8]) -> Result<(), secp256k1::Error> {
+pub fn verify_tx_signature(signature: &[u8], unsigned_trx: &[u8]) -> Result<H160, secp256k1::Error> {
     let digest = keccak256_digest(unsigned_trx);
     let message = secp256k1::Message::parse_slice(&digest)?;
 
     let recovery_id = secp256k1::RecoveryId::parse(signature[64])?;
     let signature = secp256k1::Signature::parse_slice(&signature[0..64])?;
 
     let public_key = secp256k1::recover(&message, &signature, &recovery_id)?;
-    if secp256k1::verify(&message, &signature, &public_key) {
-        Ok(())
-    } else {
-        Err(secp256k1::Error::InvalidSignature)
-    }
+    let address = keccak256_digest(&public_key.serialize()[1..]);
+    let address = H160::from_slice(&address[12..32]);
```

Comment:
You are using 2 different ways to compute ether address from the public key. Chose one of them and stick to it.

There and https://github.com/neonlabsorg/neon-evm/pull/46/files#diff-77d828dbc5822cedb23202df3c9fd48d8be70932389c3199d0b2f888387a0cffR112

Reply:
Fixed